fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/mixin/attachment/ServerWorldMixin.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.attachment;
+
+import java.util.List;
+import java.util.concurrent.Executor;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.block.entity.Spawner;
+import net.minecraft.datafixer.DataFixTypes;
+import net.minecraft.registry.RegistryKey;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.WorldGenerationProgressListener;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.random.RandomSequencesState;
+import net.minecraft.world.PersistentState;
+import net.minecraft.world.World;
+import net.minecraft.world.dimension.DimensionOptions;
+import net.minecraft.world.level.ServerWorldProperties;
+import net.minecraft.world.level.storage.LevelStorage;
+
+import net.fabricmc.fabric.impl.attachment.AttachmentPersistentState;
+
+@Mixin(ServerWorld.class)
+public class ServerWorldMixin {
+	@Inject(at = @At("TAIL"), method = "<init>")
+	private void createAttachmentsPersistentState(MinecraftServer server, Executor workerExecutor, LevelStorage.Session session, ServerWorldProperties properties, RegistryKey<World> worldKey, DimensionOptions dimensionOptions, WorldGenerationProgressListener worldGenerationProgressListener, boolean debugWorld, long seed, List<Spawner> spawners, boolean shouldTickTime, RandomSequencesState randomSequencesState, CallbackInfo ci) {
+		// Force persistent state creation
+		ServerWorld world = (ServerWorld) (Object) this;
+		var type = new PersistentState.Type<>(
+				() -> new AttachmentPersistentState(world),
+				nbt -> AttachmentPersistentState.read(world, nbt),
+				DataFixTypes.LEVEL
+		);
+		world.getPersistentStateManager().getOrCreate(type, AttachmentPersistentState.ID);
```

Comment:
Question: Does this save the nbt to the data folder even if no attachments are used?

Reply:
I think it doesn't. I haven't tested it myself, but this mixin and the persistent state class (which is supposed to not do this on purpose) was copied from Technici4an (missing a couple `@author`s that reminds me), and I recall him saying it indeed doesn't, but I can't find where.

Reply:
Testing update: as the comment on isDirty indicates, this saves data if there are persistent attachments OR if persistent attachments have been read previously. I believe the latter condition is necessary so removed persistent attachments get deleted properly. If all of the attachments are non-persistent on a fresh save file, nothing is saved.

Perhaps there is a way to make it so the file is properly deleted if no persistent attachments remain, but this doesn't seem very necessary.